LDAP-Auth/LDAPAuthenticationProviderPlugin.cs

```diff
@@ -258,6 +281,33 @@ public async Task<ProviderAuthenticationResult> Authenticate(string username, st
             return new ProviderAuthenticationResult { Username = ldapUsername };
         }
 
+        private ProfileImageFormat TryDetermineFormat(LdapAttribute value)
+        {
+            _logger.LogDebug("Trying to determine ProfileImage Format based on Attribute value");
+            var stringValue = value.StringValue;
+            if (Uri.TryCreate(stringValue, UriKind.RelativeOrAbsolute, out var uri))
+            {
+                // We can handle Url schemes as long as its http or https
+                if (uri.Scheme == Uri.UriSchemeHttps || uri.Scheme == Uri.UriSchemeHttp)
+                {
+                    _logger.LogDebug("Attribute value was valid URI and scheme was one of (http, https). ImageFormat Url");
+                    return ProfileImageFormat.Url;
+                }
+
+                throw new InvalidFormatException($"Image attribute was a valid URI but had an invalid Scheme");
```

Comment:
This is the part I am least confident in. We shouldn't support any url with non http/https schema given we are making a request via a HTTP client. If however there are other schemas that the HttpClient supports (ftp & co), this default mode will fail even if the request might have succeeded. I am content given that if they have a non http url they can just manually change the format to `Url` which ignores this logic altogether and just lets the HttpClient fail with whatever it got given but i don't know is that is behavior the maintainers want.

Some feedback from both maintainers and users would be greatly appreciated.

Reply:
> I am content given that if they have a non http url they can just manually change the format to Url

I feel like that's honestly the best way to do it. There would also need to be a change in documentation to let the user's know that though. Maybe added to the footer of the input?

Reply:
can do 👍